### ISSUE TITLE
Error importing history (msc2716)

### ISSUE THREAD

**Issue by surakin** [mautrix/whatsapp#367]
On a big and old channel I got several of this:
Error sending batch of historical messages: failed to POST /_matrix/client/unstable/org.matrix.msc2716/rooms/!xxxxxxxxxxxxxxx.org/batch_send: M_FORBIDDEN (HTTP 403): User @whatsapp_xxxxxxxxxxxxxx:gnomos.org not in room !xxxxxxxxxxxxxxxxxxx.org (None)
(the numbers matched an old phone number from one of the users, or a person who left)

**Comment by tulir** [comment]
This might be fixed with https://github.com/mautrix/go/commit/010782c6021e4544e04c3bd1a3874e283286ffe2 (which is included in the latest

**Comment by surakin** [comment]
Looks like it still happens:

Error sending batch of historical messages: failed to POST /_matrix/client/unstable/org.matrix.msc2716/rooms/!________________.org/batch_send: M_FORBIDDEN (HTTP 403): User @whatsapp_____________:.org not in room !_______________.org (None)

**Comment by tulir** [comment]
Did you use the existing portal rooms? That patch won't fix it for those, manually deleting the relevant rows in the `mx_user_profile` table row might work (or just deleting the current portal room)

**Comment by Stamper-repo** [comment]
Hi tulir!

I have had the same or similar error when syncing the history through mcs2716. Compiled the current master branch and used that binary. Double puppeting is turned on. It is a development deployment so I deleted the mautrix-whatsapp & synapse sqlite databases. I created a non-admin user and then logged that used in to Whatsapp with the `login` command. 

I see a number of expected things happen. The contacts are created and invite the synapse user. Thanks to double puppeting, the invite is automatically accepted and a room is created.

In the end, I see a number of errors in the log including the following:

```
[Dec  3, 2021 22:35:48] [Portal/**@s.whatsapp.net-**s.whatsapp.net/ERROR] Error sending batch of historical messages: failed to POST /_matrix/client/unstable/org.matrix.msc2716/rooms/!EOsxlmpjnwehDptJCI:matrix.*/batch_send: M_FORBIDDEN (HTTP 403): User @stamper:matrix.* not in room !EOsxlmpjnwehDptJCI:matrix.* (<FrozenEventV3 event_id=$IpJJv24uAzl4iut1gc5vzT01dnLodE07kN-iOo8Hs1U, type=m.room.member, state_key=@stamper1:matrix.*, outlier=False>)^[[0m
```

In other words. A completely clean setup and the error still pops up.

**Comment by Stamper-repo** [comment]
@tulir As you closed this issue, does that mean there is a fix available? I do not see any code associated with this ticket.

**Comment by sumnerevans** [comment]
We have made so many changes to the way that WA backfill works over the past couple months that it's very hard to say what exact code change fixed this issue.
